### PR TITLE
Oauth

### PR DESCRIPTION
**Pull request by AakarM1**
Oauth for google

May have to change/add client id (962811866770-6uourl66lqhb72a61263hnor0jckccgs.apps.googleusercontent.com) and client secret (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). 

TODO After publishing
1. add the final published link to the google oauth page for it to work well.
2. add the final published link to django social part with admin

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,50 +1,53 @@
...
+## Installation
 
-### Install dependencies
+*   Clone this repository
 
-```bash
-pip install -r requirements.txt
-```
+    ```bash
+    git clone https://pypi.org/project/vocabulum/
```
Comment (Fir121) [inline README.md]: Put the actual link here

File: README.md [hunk 2]
```diff
@@ -1,50 +1,53 @@
...
+    ```
+*   Install dependencies
 
-### Fix Conf URL Path
-In the virtual env: \venv\lib\site-packages\markdownx\urls.py
+    ```bash
+    pip install -r requirements.txt
```
Comment (Fir121) [inline README.md]: Add directions to make a venv just before this

File: README.md [hunk 3]
```diff
@@ -1,50 +1,53 @@
...
+*   Make migrations
 
-With:
+    ```bash
+    python manage.py makemigrations
+    ```
+* Perform migration
 
-```bash
-from django.urls import re_path as url
```
Comment (Fir121) [inline README.md:25]: this was not fixed, so don't remove it from the README, we still need to do it

File: requirements.txt
Comment (Fir121) [inline requirements.txt:1]: Add in `requests` and `PyJWT` and `django-cors-headers` latest versions. I had an issue when setting up my environment, these were missing.

Comment (Fir121) [inline requirements.txt:1]: Add a version number for django-allauth

File: quiz/templates/quiz/login.html [hunk 4]
```diff
@@ -1,6 +1,53 @@
...
+
+        .or-divider:not(:empty)::after {
+            margin-left: .25em;
+        }
+ 
+    </style>
+
     <div class="content-section">
         <form method="POST">
             {% csrf_token %}
```
Comment (Fir121) [inline quiz/templates/quiz/login.html:53]: can we not have the default form itself. Only google login allowed, consequently you can remove the sign up page and any link referring to it

File: quiz/templates/quiz/login.html [hunk 5]
```diff
@@ -11,11 +58,21 @@
...
             </div>
-        </form>
+        </form> 
+
+        <div class="or-divider">OR</div>
+
+        <div> 
+            <a href="{% provider_login_url 'google' %}">
+            <button class="btn btn-google">
+                <img src="https://playtht-website-assets.s3.amazonaws.com/img/brand/google.svg" alt="Google logo"/>Sign in with Google
```
Comment (Fir121) [inline quiz/templates/quiz/login.html]: make it `Log in` with Google

**Review by Fir121 — CHANGES_REQUESTED** [review]
All in all awesome work. Had some venv setup issues, fix that up. README updates and make it mandatory google login.

Once done will give a quick review again, resolve each point of the review when you fix it.

**Review by AakarM1 — COMMENTED** [review]
All changes made as requested

**Comment by Fir121** [comment]
Make an .ENV for the secret and client id. Load that in via python-dotenv

Add these instructions to the README

so this works properly for deployment as well.

**Comment by Fir121** [comment]
and when there are a lot of commits, we use squash and merge, it's cleaner

**Comment by Fir121** [comment]
once env is setup you can add it with your next PR. Refresh the secret as well since by now it has been accessed by 85% of the world's hacking bots

**Comment by suryavirkapur** [comment]
```
urlpatterns = [
    path('', quiz_views.home, name='home'),
    path('markdownx/', include('markdownx.urls')),
    path('admin/', admin.site.urls),
    # path('register/', quiz_views.register, name='register'),
    path('login/', auth_views.LoginView.as_view(template_name='quiz/login.html'), name='login'),
    path('logout/', auth_views.LogoutView.as_view(template_name='quiz/logout.html'), name='logout'),
    path('player/', include('player.urls')),
    path('task/', include('quiz.urls')),

    path('leaderboard/', quiz_views.leaderboard, name='leaderboard'),
    path("accounts/",include('allauth.urls')),
] + static(settings.STATIC_URL, document_root=settings.STATIC_ROOT)

    path('leaderboard/', quiz_views.leaderboard, name='leaderboard')
] + static(settings.STATIC_URL, document_root=settings.STATIC_ROOT)
```

- There are some errors. 
- Requirements.txt is missing some new deps.

**Pull request merged**